CodeEdit/WorkspaceView.swift

```diff
@@ -130,6 +133,15 @@ struct WorkspaceView: View {
                             : themeModel.selectedLightTheme
                         }
                     }
+                    .onChange(of: enableSourceControl) { newValue in
```

Comment:
I think this would read better if the variable name got updated and the conditions got switched. Also it seems like the logic is inverted. If the branch is now enabled, it should refresh the branch instead of setting it to nil.

```swift
.onChange(of: enableSourceControl) { isEnabled in
    if isEnabled {
        Task { await sourceControlManager.refreshCurrentBranch() }
    } else {
        sourceControlManager.currentBranch = nil
    }
}
```

Reply:
I am not sure I follow. It reads if git is enabled in settings we get and set the current branch otherwise we clear it out. Seems pretty straightforward to me. What did you have in mind?

Reply:
The name should definitely be updated. Also, if `enableSourceControl` changes to `true`, we now set the current branch to `nil`...

Reply:
The naming is confusing which is why I think this logic error happens. If `newValue` is true (we just enabled source control), then the check `if !newValue { }` is equivalent to `if newValue == false`, and then doing a refresh for the current branch, instead of setting the current branch to nil.

Reply:
Renaming this to `sourceControlIsEnabled`.